From 0f2217729729257bea5b62ac69ea186639075f85 Mon Sep 17 00:00:00 2001
From: Nachiket Sahasrabuddhe <nachiket_sahasrabuddhe@xyratex.com>
Date: Tue, 10 Sep 2013 11:23:57 +0530
Subject: [PATCH 27/50] bigmemalloc.mf

-Had forgotten to add lib/varr_private.h into git tree, added the same
 in this commit.
---
 lib/varr_private.h | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 lib/varr_private.h

diff --git a/lib/varr_private.h b/lib/varr_private.h
new file mode 100644
index 0000000..72aa204
--- /dev/null
+++ b/lib/varr_private.h
@@ -0,0 +1,30 @@
+/* -*- C -*- */
+/*
+ * COPYRIGHT 2013 XYRATEX TECHNOLOGY LIMITED
+ *
+ * THIS DRAWING/DOCUMENT, ITS SPECIFICATIONS, AND THE DATA CONTAINED
+ * HEREIN, ARE THE EXCLUSIVE PROPERTY OF XYRATEX TECHNOLOGY
+ * LIMITED, ISSUED IN STRICT CONFIDENCE AND SHALL NOT, WITHOUT
+ * THE PRIOR WRITTEN PERMISSION OF XYRATEX TECHNOLOGY LIMITED,
+ * BE REPRODUCED, COPIED, OR DISCLOSED TO A THIRD PARTY, OR
+ * USED FOR ANY PURPOSE WHATSOEVER, OR STORED IN A RETRIEVAL SYSTEM
+ * EXCEPT AS ALLOWED BY THE TERMS OF XYRATEX LICENSES AND AGREEMENTS.
+ *
+ * YOU SHOULD HAVE RECEIVED A COPY OF XYRATEX'S LICENSE ALONG WITH
+ * THIS RELEASE. IF NOT PLEASE CONTACT A XYRATEX REPRESENTATIVE
+ * http://www.xyratex.com/contact
+ *
+ * Original author: Anand Vidwansa <anand_vidwansa@xyratex.com>
+ * Original creation date: 12/17/2012
+ */
+
+#pragma once
+
+#ifndef __MERO_LIB_VIRTUAL_ARRAY_PRIVATE_H__
+#define __MERO_LIB_VIRTUAL_ARRAY_PRIVATE_H__
+
+M0_EXTERN void *m0_varr_buf_alloc(size_t bufsize);
+M0_EXTERN void  m0_varr_buf_free(void *buf, size_t bufsize);
+M0_EXTERN bool  m0_varr_size_is_valid(const struct m0_varr *arr);
+
+#endif /* __MERO_LIB_VIRTUAL_ARRAY_PRIVATE_H__ */
-- 
1.8.3.2

